native/src/routes/__tests__/CityNotCooperating.spec.tsx

```diff
@@ -0,0 +1,42 @@
+import { render, fireEvent, RenderAPI } from '@testing-library/react-native'
+import React from 'react'
+
+import { lightTheme } from 'build-configs/integreat/theme'
+
+import wrapWithTheme from '../../testing/wrapWithTheme'
+import CityNotCooperating from '../CityNotCooperating'
+
+jest.mock('react-i18next')
+jest.mock('styled-components', () => ({
+  ...jest.requireActual('styled-components'),
+  useTheme: () => lightTheme
+}))
+jest.mock('@react-native-clipboard/clipboard')
+jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
+
+describe('CityNotCooperating', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+  const renderCityNotCooperating = (): RenderAPI => render(<CityNotCooperating />, { wrapper: wrapWithTheme })
+
+  it('should render correctly', () => {
+    const { getByText, queryByText } = renderCityNotCooperating()
+    expect(getByText('callToAction')).toBeDefined()
+    expect(getByText('explanation')).toBeDefined()
+    expect(getByText('whatToDo')).toBeDefined()
+    expect(getByText('findOutMail')).toBeDefined()
+    expect(getByText('sendText')).toBeDefined()
+    expect(getByText('copyText')).toBeDefined()
+    expect(queryByText('textCopied')).toBeNull()
+  })
+
+  it('should copy text on button click', () => {
+    const { getByText, queryByText } = renderCityNotCooperating()
+    expect(queryByText('textCopied')).toBeNull()
+    const button = getByText('copyText')
+    fireEvent.press(button)
+    expect(getByText('textCopied')).toBeDefined()
+    expect(queryByText('copyText')).toBeNull()
```

Comment:
It would be nice to assert that the clipboard has the right content afterwards

Reply:
As we mock Clipboard, this would not really work. Also i think it doesn't make sense to test third-party functionality in our tests. But i added a check that Clipboard.setString() is called with the correct parameter.

Reply:
That's what I meant, sorry :+1: OC it's enough to assert that the third party library is called correctly.